launcher/tasks/ConcurrentTask.cpp

```diff
@@ -215,68 +213,20 @@ void ConcurrentTask::subTaskProgress(Task::Ptr task, qint64 current, qint64 tota
     task_progress->update(current, total);
 
     emit stepProgress(*task_progress);
-    updateStepProgress(*task_progress, Operation::CHANGED);
     updateState();
 
     if (totalSize() == 1) {
         setProgress(task_progress->current, task_progress->total);
     }
 }
 
-void ConcurrentTask::subTaskStepProgress(Task::Ptr task, TaskStepProgress const& task_progress)
```

Comment:
You know in retrospect this probably would be the most expensive part

Reply:
yes it was and we do not need it at all as we can just pass the step progress without any change(we interested only with the tasks that were added to the concurrentTask, the others subtasks should only be propagated for the progress dialog)